src/main/java/com/google/graphgeckos/dashboard/datatypes/BuildBotData.java

```diff
@@ -33,66 +36,89 @@
    * it in the right {@link BuildInfo} in the storage.
    * Used in {@link com.google.graphgeckos.dashboard.storage.DatastoreRepository}.
    */
+  @Transient
+  @Field(name = "commitHash")
   private String commitHash;
 
-  /* The timestamp of the build. */
+  /** 
+   * The timestamp of the build. 
+   */
+  @Field(name = "timestamp")
   private Timestamp timestamp;
 
-  /* Name of the buildbot. */
+  /**
+   * Name of the buildbot.
+   */
+  @Field(name = "name")
   private String name;
 
   /**
    * The logs of each compilation stage, stored as described by {@link Log}.
    */
+  @Field(name = "logs")
   private final List<Log> logs = new ArrayList<>();
 
   /**
    * Builder compilation status, as described by {@link BuilderStatus}.
    */
+  @Field(name = "status")
   private BuilderStatus status;
 
+  /**
+   * Only used by Spring GCP.
+   */
+  public BuildBotData() {}
+
   public BuildBotData(String commitHash, String name, List<Log> logs, BuilderStatus status) {
     this.commitHash = commitHash;
     this.name = name;
     this.logs.addAll(logs);
     this.status = status;
   }
 
-  public BuildBotData() {}
-
-  /**
-   * Returns the name of the builder bot. Cannot be null.
-   */
+  @NonNull
   public String getName() {
     return name;
   }
 
-  /**
-   * Returns the list of logs for each compilation stage. Cannot be null.
-   */
+  @NonNull
```

Comment:
nit: I'd keep the comment about this getLogs from before, as it talks about logs per compilation stage, and the "compilation stage" part is not clear just from the method signature.